src/components/Table/GridTable.tsx

```diff
@@ -279,31 +281,25 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
       // We only pass sortState to header rows, b/c non-headers rows shouldn't have to re-render on sorting
       // changes, and so by not passing the sortProps, it means the data rows' React.memo will still cache them.
       const sortProps = row.kind === "header" ? { sorting, sortState, setSortKey } : { sorting };
-      // We only pass `isCollapsed` as a prop so that the row only re-renders when it itself has
-      // changed from collapsed/non-collapsed, and not other row's entering/leaving collapsedIds.
-      // Note that we do memoized on toggleCollapsedId, but it's stable thanks to useToggleIds.
-      const isCollapsed = collapsedIds.includes(row.id);
       const RowComponent = observeRows ? ObservedGridRow : MemoizedGridRow;
 
       return (
-        <RowComponent
-          key={`${row.kind}-${row.id}`}
-          {...{
-            as,
-            columns,
-            row,
-            style,
-            rowStyles,
-            stickyHeader,
-            stickyOffset,
-            isCollapsed,
-            toggleCollapsedId,
-            // TODO: How will this effect with memoization?
-            // At least for non-nested card tables, we make this undefined so it will be fine.
-            openCards: nestedCards ? nestedCards.currentOpenCards() : undefined,
-            ...sortProps,
-          }}
-        />
+        <GridCollapseContext.Provider value={row.kind === "header" ? collapseContextAll : collapseContextRow}>
```

Comment:
What did you find as improvement from moving from 1 Context Provider to many Context Providers?

Reply:
It's now called `collapseAllContext`, that is the context provided to the header row, since it needs to know about all of the rows in order to collapse them all.

There was always many contexts, the difference is their order. It used to be:

GridRow
|-> CollapseContext

Now it is:
CollapsedContext
|-> GridRow